Added a guide for Self-Hosting Hoppscotch (#226)
hoppscotch/docs: guides/articles/self-host-hoppscotch-on-your-own-servers.mdx

---
sidebarTitle: Self-Host Hoppscotch on your own servers
title: Self-Host Hoppscotch on your own servers
description: Set up Hoppscotch on your servers for complete control and customization.
---

Self-hosting Hoppscotch gives you complete control over your API development workflow and allows you to deploy Hoppscotch in your own data center or cloud, giving you greater control over data and security.

> Self-hosted Hoppscotch comes in two variants - [Community](https://docs.hoppscotch.io/documentation/self-host/community-edition/getting-started) and [Enterprise Edition](https://docs.hoppscotch.io/documentation/self-host/enterprise-edition/getting-started), both of which can be deployed on systems that support Docker. You can host Hoppscotch on your servers, providing a workspace that is private to the individuals or teams using Hoppscotch.

Reviewer: ```suggestion
Self-hosted Hoppscotch comes in two variants - [Community](https://docs.hoppscotch.io/documentation/self-host/community-edition/getting-started) and [Enterprise Edition](https://docs.hoppscotch.io/documentation/self-host/enterprise-edition/getting-started), both of which can be deployed on systems that support Docker. You can host Hoppscotch on your servers, providing a private workspace for the individuals or teams using it.
```

> This guide covers the basics of self-hosting Hoppscotch, including the necessary configurations and settings you need to get started.

Reviewer: ```suggestion
This guide covers the basics of self-hosting Hoppscotch, including the configurations and settings needed to get started
```

## Pre-requisites

> Before you start self-hosting Hoppscotch, ensure that your system or environment meets the following requirements:

Reviewer: ```suggestion
Before you start ensure that your system or environment meets the following requirements:
```

MAILER_SMTP_ENABLE=true
MAILER_USE_CUSTOM_CONFIGS=false
MAILER_ADDRESS_FROM=<from@example.com>
MAILER_SMTP_URL=smtps://user@domain.com:pass@smtp.domain.com # used if custom mailer configs is false
# The following are used if custom mailer configs is true
MAILER_SMTP_HOST=smtp.domain.com
MAILER_SMTP_PORT=587
MAILER_SMTP_SECURE=true
MAILER_SMTP_USER=user@domain.com
MAILER_SMTP_PASSWORD=pass
MAILER_TLS_REJECT_UNAUTHORIZED=true

# Rate Limit Config
RATE_LIMIT_TTL=60 # In seconds
RATE_LIMIT_MAX=100 # Max requests per IP

#-----------------------Frontend Config------------------------------#

# Base URLs
VITE_BASE_URL=http://localhost:3000
VITE_SHORTCODE_BASE_URL=http://localhost:3000
VITE_ADMIN_URL=http://localhost:3100

# Backend URLs
VITE_BACKEND_GQL_URL=http://localhost:3170/graphql
VITE_BACKEND_WS_URL=wss://localhost:3170/graphql
VITE_BACKEND_API_URL=http://localhost:3170/v1

# Terms Of Service And Privacy Policy Links (Optional)
VITE_APP_TOS_LINK=https://docs.hoppscotch.io/support/terms
VITE_APP_PRIVACY_POLICY_LINK=https://docs.hoppscotch.io/support/privacy

# Set to `true` for subpath based access
ENABLE_SUBPATH_BASED_ACCESS=false

```

### 1. Database Configuration

> By default, Hoppscotch ships with a Docker container that has a pre-configured Postgres database. If you prefer to use your own Postgres database, either on the cloud or locally, you’ll need to set up a valid database URL with the correct credentials.

Reviewer: We dont ship with a pre-configured postgres database, why are we adding this here ?

Author: My bad

To access the admin dashboard, you’ll need to configure an OAuth provider. In the Community Edition, Hoppscotch supports:

1. Google
2. GitHub
3. Microsoft

In the Enterprise Edition, support also includes **SAML SSO, OpenID Connect, and GitHub Enterprise**.

Here’s a quick guide to registering an OAuth application with GitHub:

1. Click your profile photo in the upper-right corner and select **Settings**.
2. In the left sidebar, scroll down and click **Developer Settings**.
3. Click **OAuth Apps** in the sidebar.
4. Click **New OAuth App**.
5. Enter the required information and specify the callback URL as indicated in your configuration.
6. After registering the application, copy the Client ID and Client Secret, and add them to your environment file.

Similarly, you can follow the specific setup instructions for other OAuth providers to complete your configuration.

### 4. Subpath Access

Subpath access allows you to host multiple services under a single domain by assigning each service a specific subpath.

To enable subpath-based access, set `ENABLE_SUBPATH_BASED_ACCESS=true` (the **default** setting is **false**).

When enabled, here's how it affects your setup:

> 

Reviewer: Lets just replace this whole section and just say if subpath is enabled you can access the 3 services on the same domain but different routes and if its not enabled you access them on different ports.

Author: Done
